### ISSUE TITLE
🤖 [AUTO] Implement Strategy: spy_neural_network_simple

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#319]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 8)
**Extracted:** 2025-12-16T14:18:14.021953

---

1.  **Strategy Name**: spy_neural_network_simple
2.  **Strategy Type**: Quantitative/Machine Learning
3.  **Timeframe**: 1D (daily close to daily close)
4.  **Instruments**: SPY (S&P 500 ETF)

5.  **Entry Rules**:
    *   **Long**: Predicted next-day return from the neural network is positive.
    *   **Short**: Predicted next-day return from the neural network is negative.

6.  **Exit Rules**:
    *   **Time-based**: Hold position for one day.
    *   **Take Profit/Stop Loss**: Not explicitly mentioned, implied by one-day hold.

7.  **Indicators Required**:
    *   Past 1-day return (`ret1`)
    *   Past 2-day return (`ret2`)
    *   Past 5-day return (`ret5`)
    *   Past 20-day return (`ret20`)
    *   Feed Forward Neural Network (e.g., MATLAB's `feedforwardnet`) with a minimal architecture (e.g., 1 hidden layer with 1 neuron, `hiddenSizes=[1]`).
    *   Training involves strategies to reduce overfitting:
        *   **Retraining**: Train multiple networks (e.g., 100) with different random initial weights and data selections for training/validation. Pick the network with the lowest validation error.
        *   **Averaging**: Train multiple networks (e.g., 100) and average their predicted returns.

8.  **Risk Management**:
    *   **Position Sizing**: Buy or short a fixed dollar amount (e.g., $1).
    *   **Maximum Risk per Trade**: Not explicitly defined.
    *   **Risk-Reward Ratio**: Not explicitly defined.

9.  **Additional Notes**:
    *   Model trained on data from December 22, 2004, to September 15, 2009.
    *   Tested out-of-sample from September 16, 2009, to June 2, 2014.
    *   Out-of-sample performance (best consistent results for 1 hidden layer, 1 neuron):
        *   With retraining (100 networks): CAGR 3.2%, Sharpe (not explicitly stated for this exact metric).
        *   With averaging (100 networks): CAGR 1.9%, Sharpe (not explicitly stated for this exact metric).
    *   Overfitting is a significant concern for neural networks with limited financial data; simpler architectures tend to perform better out-of-sample.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/spy_neural_network_simple.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
